identity-core/src/common/timestamp.rs

```diff
@@ -87,6 +86,30 @@ impl Timestamp {
     }
     Ok(Self(offset_date_time))
   }
+
+  /// Computes `self + duration`
+  ///
+  /// # Errors
+  /// Errors if the operation leads to a timestamp not in the valid range for [RFC 3339](https://tools.ietf.org/html/rfc3339).
+  pub fn try_add(self, duration: Duration) -> Result<Self> {
+    self
+      .0
+      .checked_add(duration.0)
+      .ok_or_else(|| time::error::Error::Format(time::error::Format::InvalidComponent("invalid year")).into())
```

Comment:
Not sure if we should return an error rather than an option. On a related point, we could change the timestamp error variant to have an optional source so we can return just a message here.

E.g.
```rust
  /// Caused by attempting to parse or perform arithmetic with a `Duration` that would lead to an invalid `Timestamp`.
  #[error("Invalid Timestamp: {0}")]
  InvalidTimestamp(&'static str, #[source] Option<time::error::Error>),
```

Either way, I think the error message should be more descriptive.

```suggestion
  pub fn try_add(self, duration: Duration) -> Result<Self> {
    self
      .0
      .checked_add(duration.0)
      .ok_or_else(|| time::error::Error::Format(time::error::Format::InvalidComponent("addition exceeded maximum RFC3999 timestamp")).into())
```

Reply:
I think returning an `Option` and changing the name to `checked_add` is more in line with how the rest of the ecosystem deals with arithmetic overflow. This also enables us to implement the `CheckedAdd` trait from [num_traits](https://docs.rs/num-traits/0.2.14/num_traits/ops/checked/trait.CheckedAdd.html) in the future.  

Moreover I expect the real world cases where one actually ends up with an RFC3339 incompatible timestamp from these operations to be extremely rare so I would prefer not to pollute the API with more (or more complicated) error conditions.